one_fm/utils.py

```diff
@@ -782,96 +826,160 @@ def create_leave_allocation(employee, policy_detail, leave_type_details, from_da
-                leave_days = 0
-                if leave_type.one_fm_paid_sick_leave_type_dependent:
-                    leave_days += get_leaves_for_period(allocation.employee, leave_type.one_fm_paid_sick_leave_type_dependent, allocation.from_date, allocation.to_date)
-                else:
-                    leave_type_list = frappe.get_all('Leave Type', filters= {'one_fm_is_paid_sick_leave': 1}, fields= ["name"])
-                    for lt in leave_type_list:
-                        leave_days += get_leaves_for_period(allocation.employee, lt.name, allocation.from_date, allocation.to_date)
-
-                if leave_days > 0:
-                    percent_of_reduction = 0
-                    allocation_reduction = sorted(leave_type.one_fm_annual_leave_allocation_reduction, key=lambda x: x.number_of_paid_sick_leave)
-                    for alloc_reduction in allocation_reduction:
-                        if alloc_reduction.number_of_paid_sick_leave <= leave_days:
-                            percent_of_reduction = alloc_reduction.percentage_of_allocation_reduction
-                    if percent_of_reduction > 0:
-                        new_leaves_allocated = daily_allocation - daily_allocation * (percent_of_reduction/100)
-
-            allocation.new_leaves_allocated = allocation.new_leaves_allocated+new_leaves_allocated
-            allocation.total_leaves_allocated = allocation.new_leaves_allocated+allocation.unused_leaves
-            allocation.save()
-
-            ''' Delete old ledger entry'''
-            frappe.db.sql("""DELETE FROM `tabLeave Ledger Entry`
-                WHERE `transaction_name`=%s""", (allocation.name))
-
-            ''' Create new ledger entry'''
-            ledger = frappe._dict(
-                doctype='Leave Ledger Entry',
-                employee=allocation.employee,
-                leave_type=leave_type.name,
-                transaction_type='Leave Allocation',
-                transaction_name=allocation.name,
-                leaves = allocation.total_leaves_allocated,
-                from_date = allocation.from_date,
-                to_date = allocation.to_date,
-                is_carry_forward=leave_type.is_carry_forward,
-                is_expired=0,
-                is_lwp=0
-            )
-            frappe.get_doc(ledger).submit()
-
+    return frappe.db.sql(query.format(getdate(date)), as_dict=True)
+
+def is_employee_allowed_to_avail_increment_existing_allocation(allocation, leave_type):
+    '''
+        Function is used to check whether the employee is allowed to avail incrementing the leave allocation
+        args:
+            allocation: Object of Leave Allocation
+            leave_type: Object of Leave Type
+        return: Boolean
+    '''
+    allow_allocation = True
+
+    # Check if employee absent today then not allow annual leave allocation for today
+    is_absent = frappe.db.sql("""select name, status from `tabAttendance` where employee = %s and
```

Comment:
@pjamsheer , marking `absent` happened at the end of the shift, so he might be absent today, but still, it is not set as absent in the system because the shit didn't end yet.
means you might get faulty data based on when your process will run.

Reply:
If he/she is absent or not present without a leave application then He/She allowed to have an allocation for that day. So there will not be any issue comes

Reply:
👍